src/WeakRefStrings.jl

```diff
@@ -191,6 +193,15 @@ const MISSING_OFFSET = typemax(UInt64)-1
 Base.size(a::StringArray) = size(a.offsets)
 Base.IndexStyle(::Type{<:StringVector}) = IndexLinear()
 
+function DataAPI.refarray(a::WeakRefStrings.StringArray{T}) where {T}
+    S = Union{WeakRefStrings.WeakRefString{UInt8}, typeintersect(T, Missing)}
+    convert(WeakRefStrings.StringArray{S}, a)
```

Comment:
I don't understand why this method calls `convert`. I thought `refarray` should always return an array without making a copy since it's there for performance?

Reply:
The trick is that it actually calls this method: https://github.com/JuliaData/WeakRefStrings.jl/blob/master/src/WeakRefStrings.jl#L201 which does not allocate, but change the `eltype` so that indexing no longer allocate the string. Maybe I should use `StringArray{S, ndims(a)}(a.buffer, a.offsets, a.lengths)` directly, so that it's clear there are no allocations.

Reply:
OK, that's fine then.